### ISSUE TITLE
Java indexer: emit `overrides` edges for direct method overrides through a superclass [phab:D1915]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2358]
_Created by **schroederc** at 2017-08-29 19:01:13_: 

```
class A { void m() {}; }
class B extends A {}
class C extends B { void m() {}; }
```

In the above case, the indexer will now state that C.m *directly*
overrides A.m since there is no intermediary B.m method.

Before, C.m was indirectly overriding A.m, but the relationship was
strange since C.m had no direct overrides.  This also confused the
post-processor such that C.m and A.m would have *no* overrides
relationship in the final serving data.

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2017-08-29 19:35:20_

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2017-08-29 19:36:19_

**Comment by kythe** [comment]
_**jrtom** commented inline (context lost during migration) at 2017-08-29 19:39:31_:

Is there a good reason for this class to exist at all?  Its only contents are two static methods, each of which is only used in KytheTreeScanner.  Would seem to make more sense to just put those methods in KTS instead.

**Comment by kythe** [comment]
_**jrtom** commented inline (context lost during migration) at 2017-08-29 19:39:31_:

Having trouble parsing "a method could be found to both a..."

Maybe either "a method could be both a" or "a method could be found to be both a"

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-08-29 20:43:00_:

Doc

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2017-08-29 20:45:05_:

It's just legacy structure.  It can changed in a different revision.

**Comment by kythe** [comment]
_**schroederc** closed via commit 8cfa16d5af9d442e0c21e77fc7eafd788eb65852 at 2017-08-29 20:45:44_
